src/ChatClient/ChatClientFactory.cs

```diff
@@ -43,14 +43,33 @@ public static IChatClient CreateCopilotChatClientWithGitHubToken()
 
         // Get the Copilot token using the GitHub token
         var helper = new GitHubCopilotHelper();
-        var copilotToken = helper.GetCopilotTokenSync(githubToken);
+        var tokenDetails = helper.GetCopilotTokenDetailsSync(githubToken);
         
-        if (string.IsNullOrEmpty(copilotToken))
+        if (string.IsNullOrEmpty(tokenDetails.token))
         {
             throw new EnvVarSettingException("Failed to get a valid Copilot token from GitHub. Please run 'chatx github login' to authenticate.");
         }
 
-        var options = InitOpenAIClientOptions(endpoint, $"Bearer {copilotToken}");
+        // Create options with the initial auth header
+        var options = InitOpenAIClientOptions(endpoint, $"Bearer {tokenDetails.token}");
+        
+        // Create the refresh policy with a callback that updates both token and expiration
+        CopilotTokenRefreshPolicy refreshPolicy = null!;
```

Comment:
variable declarations in this codebase are typically `var` not `ActualType`... Thus, use `var refreshPolicy = ...`.